public/app/features/explore/TraceView/components/TraceTimelineViewer/SpanDetail/index.tsx

```diff
@@ -214,7 +214,12 @@ export default function SpanDetail(props: SpanDetailProps) {
                 type: 'log',
                 location: 'spanDetails',
               });
-              logLinks?.[0].onClick?.(event);
+
+              if (logLinks?.[0].onClick) {
+                logLinks?.[0].onClick?.(event);
+              } else {
+                window.location.href = logLinks?.[0].href;
```

Comment:
this causes a full page reload,  never do this to navigate.  

If it's a link why are we doing this?? just let the default link click handler pass this off to react router. 

If you really need to navigate manually use locationService.push

Reply:
Yeah the `DataLinkButton` actually creates a normal link with correct `href`. The root cause here seems when we added the reportInteraction here https://github.com/grafana/grafana/commit/2b6a5e2b3db73ec0f35589fe7b9a44da5a8b0896#diff-294dea7ed5321a98ecbeaad2d2377b1316b0b29bdb3b7d579ffbb2d4fc83c75cR203.

Because in the dataLinkButton we check if onClick exists and if it does we assume it does the actual handling and stops propagation of normal navigation (this is needed otherwise in explore we would both open split but right after navigate to new page making the openSplit function no-op).

So not sure if manually navigating in the onClick here makes sense or if it would be better to alter the DataLinkButton to maybe take some other prop just for the interaction reporting (that way we will know that it alone won't handle navigation) because otherwise you don't know what is inside the onClick and whether `event.preventDefault()` should be called or not.

Reply:
Out of curiosity - what is used under the hood to send tracking events? Do we use `sendBeacon()` to avoid the situation where navigating away with page reload would cancel pending requests to send analytics from the previous page?

Reply:
I fixed it to use the locationService and added a clarifying comment. I think the UI team might prefer to keep feature tracking code separate from UI elements and I'd rather not drive that conversation as part of this bug, so hopefully the correct redirect logic and a comment on why we need it is sufficient to move this forward.